spec/controllers/api/v1/application_users_controller_spec.rb

```diff
@@ -46,6 +45,29 @@
     end
   end
 
+  describe "username returns" do
+    it "a single result when username matches" do
+      api_get :username, untrusted_application_token, parameters: { username: 'foo_bb' }
+      expect(response.code).to eq('200')
```

Comment:
Actually `foo_bb` does belong to `untrusted_application` :) Your point is well made though, that only succeeded though dumb luck since I wasn't aware of that issue.  I've added more specs to check that an application can't search for users that they don't have.